packages/dds/tree/src/test/simple-tree/schemaFactory.spec.ts

```diff
@@ -327,6 +328,18 @@ describe("schemaFactory", () => {
 			assert.equal(MyList.identifier, `test.Array<["${factory.number.identifier}"]>`);
 		});
 
+		it("Scoped", () => {
+			const factory = new SchemaFactory("test-scope");
+			// We specified a scope in the factory, so it should be part of the type signature of the created object
+			const foo: TreeNodeSchemaClass<"test-scope.foo"> = factory.object("foo", {});
```

Comment:
add an assert that then runtime behavior actually works as well. Also I think this is more robust (ex: "any" would fail)
```suggestion
			const foo = factory.object("foo", {}).identifier;
			type _check = requireTrue<areSafelyAssignable<"test-scope.foo", typeof foo>>;
			assert.equal(foo , "test-scope.foo");
```

Reply:
same for Unscoped